src/plugin/kysely-plugin.ts

```diff
@@ -13,21 +13,25 @@ export interface KyselyPlugin {
    * If you need to pass some query-related data between this method and `transformResult` you
    * can use a `WeakMap` with {@link PluginTransformQueryArgs.queryId | args.queryId} as the key:
    *
-   * ```ts
-   * const plugin = {
-   *   data: new WeakMap<QueryId, SomeData>(),
-   *
-   *   transformQuery(args: PluginTransformQueryArgs): RootOperationNode {
-   *     this.data.set(args.queryId, something)
-   *     return args.node
-   *   },
-   *
-   *   transformResult(args: PluginTransformResultArgs): QueryResult<UnknownRow> {
-   *     const data = this.data.get(args.queryId)
-   *     return args.result
-   *   }
-   * }
-   * ```
+```ts
+import { KyselyPlugin, PluginTransformQueryArgs, PluginTransformResultArgs } from 'kysely';
+
+class MyPlugin implements KyselyPlugin {
+  private data = new WeakMap<QueryId, SomeData>();
+
+  transformQuery(args: PluginTransformQueryArgs) {
+    this.data.set(args.queryId, something);
+    return args.node;
+  }
+
+  async transformResult(args: PluginTransformResultArgs) {
+    const data = this.data.get(args.queryId);
+    return args.result;
+  }
+}
+
+const myPlugin = new MyPlugin();
+```
```

Comment:
Let's keep it focused. 

The intention is to show how to pass data from `transformQuery` to `transformResult`.
Things like imports or how to instantiate the class are noise here.

It is also missing some `*`s at the start of each line..

```suggestion
```ts
class MyPlugin implements KyselyPlugin {
  private data = new WeakMap<QueryId, SomeData>();

  transformQuery(args: PluginTransformQueryArgs) {
    this.data.set(args.queryId, something);
    return args.node;
  }

  async transformResult(args: PluginTransformResultArgs) {
    const data = this.data.get(args.queryId);
    return args.result;
  }
}
```